lib/src/transport/receiver-xhr.dart

```diff
@@ -3,30 +3,33 @@ part of sockjs_client;
  class XhrReceiver extends Receiver {
 
    AbstractXHRObject xo = null;
+   int buf_pos = 0;
 
    XhrReceiver(url, AjaxObjectFactory xhrFactory, {bool noCredentials}) {
-    var buf_pos = 0;
-
     xo = xhrFactory('POST', url, noCredentials: noCredentials);
     xo.onChunk.listen((e){
         if (e.status != 200) return;
-        while (true) {
-            var buf = e.text.substring(buf_pos);
-            var p = buf.indexOf('\n');
-            if (p == -1) break;
-            buf_pos += p+1;
-            var msg = buf.substring(0, p);
-            dispatch(new MessageEvent(msg));
-        }
+        readAndBroadcastMessage(e);
     });
     xo.onFinish.listen((e) {
-        dispatch(new StatusEvent("chunk", e.status, e.text));
```

Comment:
this event doesn't make sense, the XHRObject is the only one who should be emitting chunk events. The receiver should only emits CloseEvents ands MessageEvents. No one would even be listening to this.